src/tests/conformance/messages.rs

```diff
@@ -372,6 +377,125 @@ async fn correctly_lists_peers() {
+    //  1. unlimited queries i.e. stop_hash = 0
+    //  2. range queries i.e. stop_hash = i
+    //  3. a forked chain (we submit a header which doesn't match the chain)
+    //
+    // Test procedure:
+    //  1. Create a node and seed it with the testnet chain
+    //  2. Establish a peer node
+    //  3. For each test case:
+    //      a) send GetHeaders
+    //      b) receive Headers
+    //      c) assert headers received match expectations
+    //
+    // The test currently fails for both Zebra and zcashd.
+    //
+    // Current behaviour:
+    //
+    //  zcashd: Fails for range queries where the head of the chain equals the stop hash. We expect to receive an empty set,
+    //          but instead we get header [i+1] (which exceeds stop_hash).
+    //
+    //  zebra: does not support seeding as yet, and therefore cannot perform this test.
+
+    let (zig, node_meta) = read_config_file();
+
+    // Create a node with knowledge of the initial three testnet blocks
+    let mut node = Node::new(node_meta);
+    node.initial_action(Action::SeedWithTestnetBlocks(zig.new_local_addr()))
+        .start()
+        .await;
+
+    // block headers and hashes
+    let expected = vec![
+        Block::testnet_genesis().header,
+        Block::testnet_1().header,
+        Block::testnet_2().header,
+    ];
+    let hashes = expected
+        .iter()
+        .map(|header| header.double_sha256().unwrap())
+        .collect::<Vec<_>>();
+
+    // locator hashes are in stored in reverse order
+    let locator = vec![
+        vec![hashes[0].clone()],
+        vec![hashes[1].clone(), hashes[0].clone()],
+        vec![hashes[2].clone(), hashes[1].clone(), hashes[0].clone()],
+    ];
+
+    // Establish a peer node
+    let mut stream = initiate_handshake(node.addr()).await.unwrap();
+    let filter = MessageFilter::with_all_auto_reply();
+
+    // Query for all blocks from i onwards (stop_hash = [0])
+    for i in 0..expected.len() {
+        Message::GetHeaders(LocatorHashes::new(locator[i].clone(), Hash::zeroed()))
+            .write_to_stream(&mut stream)
+            .await
+            .unwrap();
+
+        match filter.read_from_stream(&mut stream).await.unwrap() {
+            Message::Headers(headers) => assert_eq!(
+                headers.headers,
+                expected[(i + 1)..],
+                "test for Headers([{}..])",
+                i
+            ),
+            messsage => panic!("Expected Headers, but got: {:?}", messsage),
+        }
+    }
+
+    // Query for all possible valid ranges
+    let ranges: Vec<(usize, usize)> = vec![(0, 0), (0, 1), (0, 2), (1, 1), (1, 2), (2, 2)];
+    for (start, stop) in ranges {
+        Message::GetHeaders(LocatorHashes::new(
+            locator[start].clone(),
+            hashes[stop].clone(),
+        ))
+        .write_to_stream(&mut stream)
+        .await
+        .unwrap();
+
+        // requried to handle cases where start and stop are the same
```

Comment:
Could you extend the comment to include why there's a need to shift the expected `start_i` index checked in the response if the `start` and `stop` indices are the same?

Reply:
Looking at this with a fresh(er) mind, I no longer think its required at all.